patches/api/0333-Added-getHostname-to-AsyncPlayerPreLoginEvent.patch

```diff
@@ -0,0 +1,68 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: MCMDEV <john-m.1@gmx.de>
+Date: Fri, 24 Sep 2021 17:59:23 +0200
+Subject: [PATCH] Added getHostname to AsyncPlayerPreLoginEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
+index 694a81769076ea58aae9f14f076ab80c9952c957..eb44bdedca85fa66921192c5c4a124c6e43fbe32 100644
+--- a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
++++ b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
+@@ -22,6 +22,7 @@ public class AsyncPlayerPreLoginEvent extends Event {
+     private final InetAddress ipAddress;
+     private final InetAddress rawAddress; // Paper
+     //private UUID uniqueId; // Paper - Not used anymore
++    private final String hostname; // Paper
+ 
+     @Deprecated
+     public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress) {
+@@ -66,6 +67,8 @@ public class AsyncPlayerPreLoginEvent extends Event {
+         this(name, ipAddress, ipAddress, uniqueId, profile);
+     }
+ 
++    // Paper start
++    @Deprecated
+     public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress, @NotNull final InetAddress rawAddress, @NotNull final UUID uniqueId, @NotNull PlayerProfile profile) {
+         super(true);
+         this.profile = profile;
+@@ -78,6 +81,20 @@ public class AsyncPlayerPreLoginEvent extends Event {
+         //this.uniqueId = uniqueId; // Paper - Not used anymore
+     }
+ 
++    public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress, @NotNull final InetAddress rawAddress, @NotNull final UUID uniqueId, @NotNull PlayerProfile profile, String hostname) {
++        super(true);
++        this.profile = profile;
++        // Paper end
++        this.result = Result.ALLOWED;
++        this.message = net.kyori.adventure.text.Component.empty(); // Paper
++        //this.name = name; // Paper - Not used anymore
++        this.ipAddress = ipAddress;
++        this.rawAddress = rawAddress; // Paper
++        //this.uniqueId = uniqueId; // Paper - Not used anymore
++        this.hostname = hostname;
++    }
++    // Paper end
++
+     /**
+      * Gets the current result of the login, as an enum
+      *
+@@ -261,6 +278,19 @@ public class AsyncPlayerPreLoginEvent extends Event {
+         return profile.getId(); // Paper
+     }
+ 
++    // Paper start
++    /**
++     * Gets the hostname that the player used to connect to the server, or
++     * blank if unknown
++     *
++     * @return The hostname
++     */
++    @NotNull
```

Comment:
the hostname can be null, because of the constructor without a hostname.

Reply:
Yeah, but that should be fixed(able) when the merger goes through and reformats the new ctor to not take up so much diff and make the old one call the new one with an empty string.